src/components/Form/ProfileForm.tsx

```diff
@@ -75,30 +77,38 @@ function ProfileForm({ callback }: { callback?: () => void }) {
   ) => {
     if (user && walletClient && publicClient) {
       try {
-        const cid = await postToIPFS(
-          JSON.stringify({
-            title: values.title,
-            role: values.role,
-            image_url: values.image_url,
-            video_url: values.video_url,
-            name: values.name,
-            about: values.about,
-            skills: values.skills,
-            web3mailPreferences: user.description?.web3mailPreferences,
-          }),
-        );
+        let cid = await tlClient?.profile.upload({
+          title: values.title,
+          role: values.role,
+          image_url: values.image_url,
+          video_url: values.video_url,
+          name: values.name,
+          about: values.about,
+          skills: values.skills,
+        });
+
         let tx;
+        console.log("SDK consumer delegate: ", isActiveDelegate)
+
         if (isActiveDelegate) {
           const response = await delegateUpdateProfileData(chainId, user.id, user.address, cid);
           tx = response.data.transaction;
         } else {
-          tx = await walletClient.writeContract({
-            address: config.contracts.talentLayerId,
-            abi: TalentLayerID.abi,
-            functionName: 'updateProfileData',
-            args: [user.id, cid],
-            account: address,
-          });
+          const res = await tlClient?.profile.update(
+            {
+              title: values.title,
+              role: values.role,
+              image_url: values.image_url,
+              video_url: values.video_url,
+              name: values.name,
+              about: values.about,
+              skills: values.skills,
+            },
+            user.id
+          );  
+
+           tx = res?.tx || "";
```

Comment:
that's a bit weird, we need to find a clean solution to remove that. 
res.tx and res.cid must always be defined here, else there must be an exception throw by the update itself